tests/MongoDB.Driver.Tests/MongoClientTests.cs

```diff
@@ -206,6 +210,33 @@ public void UsesSameClusterWhenReadPreferenceTagsAreTheSame()
             databaseNames.Should().Equal(operationResult["databases"].AsBsonArray.Select(record => record["name"].AsString));
         }
 
+        // will probably need to move this test?
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void ReadPreference_should_not_be_sent_to_standalone_server(
```

Comment:
When I rolled back the changes from this ticket, this test has still passed. So, probably it should be investigated.

Reply:
Noted, I will look into this further. It may be that the driver already handles the standalone case.

Reply:
A good local test might be to remove requirement for a standalone and then run the test against a replicaset or a sharded cluster (the test should fail since we SHOULD be sending the read preference in these cases).  I think the issue might be that the test should be asserting that the command contained within the CommandStartedEvent doesn't contain a read preference.

Reply:
Probably we can test something like:

    if (serverType == `standalone`)
    {
         // readpreference is expected to be null
    } 
    else
    {
        // readpreference is still presented
    }

Reply:
The conditional above looks fine to me (we'd need to move the Require statement ofc).

Reply:
@DmitryLukyanov the conditional you suggested works locally but some Evergreen patches are unhappy about with it - it looks like those tests aren't finding readPreference when it should be there. I'm not quite sure how to fix it on my end.